File: app/src/main/java/com/example/myschoolwearclient/ui/activity/CheckoutActivityViewModel.java
package com.example.myschoolwearclient.ui.activity;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.myschoolwearclient.firebaseClasses.AddressItem;
import com.example.myschoolwearclient.repo.AddressRepository;

public class CheckoutActivityViewModel extends ViewModel {

    private AddressRepository addressRepository = new AddressRepository();

    public LiveData<AddressItem> currentAddress = addressRepository.userAddress;

    public void updateAddress(AddressItem addressItem){
        addressRepository.updateUserAddress(addressItem);
    }

    public void getAddress(String userId){
        addressRepository.getUserAddress(userId);
    }

}

File: app/src/main/java/co/in/plantonic/ui/activity/CheckoutActivityViewModel.kt
package co.`in`.plantonic.ui.activity

import androidx.lifecycle.ViewModel
import co.`in`.plantonic.repo.AddressRepository
import androidx.lifecycle.LiveData
import co.`in`.plantonic.firebaseClasses.AddressItem
import co.`in`.plantonic.repo.BlueDartRepository

class CheckoutActivityViewModel : ViewModel() {
    private val addressRepository = AddressRepository()
    var currentAddress: LiveData<AddressItem> = addressRepository.userAddress
    fun updateAddress(addressItem: AddressItem?) {
        addressRepository.updateUserAddress(addressItem)
    }

    fun getAddress(userId: String?) {
        addressRepository.getUserAddress(userId)
    }

    private val repo = BlueDartRepository()
    val isLoading: LiveData<Boolean> = repo.isLoading
    val errorMessage: LiveData<String> = repo.errorMessage
    val isPinCodeAvailable = repo.isPinCodeAvailable
    fun checkIfPinCodeAvailable(pincode: String?) {
        repo.checkIsPinCodeAvailable(pincode!!)
    }
}